src/nvim/ops.c

```diff
@@ -3644,6 +3644,10 @@ int do_join(size_t count,
    */
   for (t = (linenr_T)count - 1;; t--) {
     cend -= currsize;
+	if(cend == NULL || curr == NULL){
```

Comment:
@Zman94 That part of the documentation is referring to `git rebase -i`.  `exec` is one of the recognized keywords for the rebase buffer which tells git to run a command (in this case `make -C build unittest`) before moving on to the next commit.

If you instead ran this manually from your shell, then it's expected that the terminal exited.  The `exec` builtin in the shell replaces the shell process with the subsequent command.  Therefore, when make is done running, there's nothing running in the terminal anymore.